### PR TITLE
out_datadog: Document `dd_hostname` configuration option

### PR DESCRIPTION
**Pull request by jszwedko**
Added by https://github.com/fluent/fluent-bit/pull/8988

### CODE REVIEW

File: pipeline/outputs/datadog.md [hunk 1]
```diff
@@ -24,7 +24,8 @@ Before you begin, you need a [Datadog account](https://app.datadoghq.com/signup)
...
+| dd_hostname     | The host that emitted logs should be associated with. If unset, Datadog will expect the host to be set as `host`, `hostname`, or `syslog.hostname` attributes. See Datadog Logs preprocessor documentation for up-to-date recognized attributes.                                                                                                                                                                                                                                                                                                                       |                                  |
```
Comment (sudomateo) [inline pipeline/outputs/datadog.md:28]: Is there a link we can use to link the "Datadog Logs preprocessor documentation"?

Comment (cnorris-cs) [inline pipeline/outputs/datadog.md:28]: The previously mentioned question is still applicable, but this suggestion updates the content to fit the new style guide, including having values be listed in ticks and not single quotes, and to not have empty cells in tables. It also makes a few wording changes for clarity (hopefully the content is still correct based on those).

```suggestion
| dd_message_key  | By default, the plugin searches for the key `log` and remaps the value to the key `message`. If the property is set, the plugin searches the property name key. | _none_ |
| dd_hostname     | The host the emitted logs should be associated with. If unset, Datadog expects the host to be set with `host`, `hostname`, or `syslog.hostname` attributes. See Datadog Logs preprocessor documentation for recognized attributes. | _none_ |
```

Comment (jszwedko) [reply pipeline/outputs/datadog.md]: Thanks @cnorris-cs ! The suggestion looks good to me. I applied it.

Comment (jszwedko) [reply pipeline/outputs/datadog.md]: 👍 added a link in [c4d9263](https://github.com/fluent/fluent-bit-docs/pull/1401/commits/c4d9263d8836e599f7a186acf482b427de16a46a)

Comment (sudomateo) [reply pipeline/outputs/datadog.md]: Nice! Thank you.

**Review by sudomateo — APPROVED** [review]
Thank you for working on this. I'm not a maintainer but I just wanted to reset the stale bot.

**Comment by jszwedko** [comment]
@fluent/fluent-bit-maintainers do you think we could have this merged?

**Comment by jszwedko** [comment]
@cnorris-cs do you think we could merge this?

**Comment by esmerel** [comment]
@fluent/fluent-bit-maintainers - any objections?

**Review by lecaros — APPROVED** [review]

**Pull request merged**